### ISSUE TITLE
Include ignored in patch requests

### ISSUE THREAD

**Issue by Ouadie** [wopian/kitsu#433]
Patching a resource with **include** is not working and the include gets added to the body as a field. 
It is not possible to add include as: `patch /v1/users/1?include=resource`
The include params is not used here -> 
https://github.com/wopian/kitsu/blob/8811addad1e0d7752ee1b0acc52cbd72bf485c86/packages/kitsu/src/index.js#L252

**Comment by wopian** [comment]
Looks like I overlooked the note right at the bottom of each query type:

> Note: This section applies to any endpoint that responds with a resource collection as primary data, regardless of the request type.

Two options for the parameter structure. Either way it has to be a breaking change as I don't want the parameters to be inconsistent between different functions - i.e a non-breaking implementation would have:
- `GET` as: model, **params** and headers
- `PATCH` as: model, body, headers and **params**
- `POST` as: model, body, headers and **params**
- `DELETE` as: model, id, headers and **params**

---

### Option 1

Preferred option as its cleaner and avoids the problems of multiple optional parameters.

#### Parameters and headers set

```js
api.get('anime', { 
  params: { filter: { seasonYear: 2020 } }, 
  headers: { myHeader: 'something' } 
})

api.patch('anime', body, { 
  params: { filter: { seasonYear: 2020 } }, 
  headers: { myHeader: 'something' } 
})

api.post('anime', body, { 
  params: { filter: { seasonYear: 2020 } }, 
  headers: { myHeader: 'something' } 
})

api.delete('anime', '1', {
  params: { filter: { seasonYear: 2020 } }, 
  headers: { myHeader: 'something' } 
}
```

#### Only headers set

```js
api.get('anime', { 
  headers: { myHeader: 'something' } 
})

api.patch('anime', body, { 
  headers: { myHeader: 'something' } 
})

api.post('anime', body, { 
  headers: { myHeader: 'something' } 
})

api.post('anime', '1', { 
  headers: { myHeader: 'something' } 
})
```

### Option 2

Match existing parameter structure. Pretty ugly if you need custom headers but not parameters.

It worked for `GET` as you'd be using parameters in almost all requests to reduce the attributes returned anyway.

#### Parameters and headers set

```js
api.get('anime', { filter: { seasonYear: 2020 } }, { myHeader: 'something' })
api.patch('anime', body, { filter: { seasonYear: 2020 } }, { myHeader: 'something' })
api.post('anime', body, { filter: { seasonYear: 2020 } }, { myHeader: 'something' })
api.delete('anime', '1', { filter: { seasonYear: 2020 } }, { myHeader: 'something' })
```

#### Only headers set

```js
api.get('anime', undefined, { myHeader: 'something' })
api.patch('anime', body, undefined, { myHeader: 'something' })
api.post('anime', body, undefined, { myHeader: 'something' })
api.delete('anime', '1', undefined, { myHeader: 'something' })
```

**Comment by wopian** [comment]
I will be using option 1 for a `10.x` release - for now you can use the [`request`] function to do patch requests with JSON:API query parameters.

[`request`]:https://github.com/wopian/kitsu/tree/master/packages/kitsu#request

**Comment by wopian** [comment]
Added in https://github.com/wopian/kitsu/commit/f98eef9d559909e1fa7787e123b0340ea33a5900 and associated commits.

**Comment by wopian** [comment]
Released in `10.0.0-alpha.0`

```js
api.patch('anime', body, { 
  params: { include: 'resource' } 
}) // anime?include=resource
```
